libraries/rush-lib/src/cli/RushXCommandLine.ts

```diff
@@ -45,129 +47,139 @@ interface IRushXCommandLineArguments {
+        showVerbose: false
+      });
+      const eventHooksManager: EventHooksManager | undefined = rushConfiguration
+        ? new EventHooksManager(rushConfiguration)
+        : undefined;
+
+      const ignoreHooks = process.env.INSIDERUSH === '1';
+      eventHooksManager?.handle(Event.preRushx, false /* isDebug */, ignoreHooks);
+      RushXCommandLine._launchRushXInternal(launcherVersion, options, rushConfiguration, args);
+      eventHooksManager?.handle(Event.postRushx, false /* isDebug */, ignoreHooks);
+    } catch (error) {
+      console.log(colors.red('Error: ' + (error as Error).message));
+    }
   }
 
   /**
    * @internal
    */
-  public static _launchRushXInternal(launcherVersion: string, options: ILaunchRushXInternalOptions): void {
+  public static _launchRushXInternal(
+    launcherVersion: string,
+    options: ILaunchRushXInternalOptions,
+    rushConfiguration: RushConfiguration | undefined,
+    args: IRushXCommandLineArguments
+  ): void {
     // Node.js can sometimes accidentally terminate with a zero exit code  (e.g. for an uncaught
     // promise exception), so we start with the assumption that the exit code is 1
     // and set it to 0 only on success.
     process.exitCode = 1;
 
-    const args: IRushXCommandLineArguments = this._getCommandLineArguments();
-
     if (!args.quiet) {
       RushStartupBanner.logStreamlinedBanner(Rush.version, options.isManaged);
     }
+    // Are we in a Rush repo?
+    NodeJsCompatibility.warnAboutCompatibilityIssues({
+      isRushLib: true,
+      alreadyReportedNodeTooNewError: !!options.alreadyReportedNodeTooNewError,
+      rushConfiguration
+    });
+
+    // Find the governing package.json for this folder:
+    const packageJsonLookup: PackageJsonLookup = new PackageJsonLookup();
+
+    const packageJsonFilePath: string | undefined = packageJsonLookup.tryGetPackageJsonFilePathFor(
+      process.cwd()
+    );
+    if (!packageJsonFilePath) {
+      console.log(colors.red('This command should be used inside a project folder.'));
+      console.log(
+        `Unable to find a package.json file in the current working directory or any of its parents.`
+      );
+      return;
+    }
 
-    try {
-      // Are we in a Rush repo?
-      const rushConfiguration: RushConfiguration | undefined = RushConfiguration.tryLoadFromDefaultLocation({
-        showVerbose: false
-      });
-      NodeJsCompatibility.warnAboutCompatibilityIssues({
-        isRushLib: true,
-        alreadyReportedNodeTooNewError: !!options.alreadyReportedNodeTooNewError,
-        rushConfiguration
-      });
-
-      // Find the governing package.json for this folder:
-      const packageJsonLookup: PackageJsonLookup = new PackageJsonLookup();
-
-      const packageJsonFilePath: string | undefined = packageJsonLookup.tryGetPackageJsonFilePathFor(
-        process.cwd()
+    if (rushConfiguration && !rushConfiguration.tryGetProjectForPath(process.cwd())) {
```

Comment:
Sorry for the big formatting change here.  The main thing to notices is that the args and rushConfiguration are calculated by the caller now.